src/adapter/src/coord/statement_logging.rs

```diff
@@ -78,11 +89,48 @@ impl StatementLogging {
             executions_begun: BTreeMap::new(),
             unlogged_sessions: BTreeMap::new(),
             reproducible_rng: rand_chacha::ChaCha8Rng::seed_from_u64(42),
+            pending_statement_execution_events: Vec::new(),
+            pending_prepared_statement_events: Vec::new(),
+            pending_session_events: Vec::new(),
         }
     }
 }
 
 impl Coordinator {
+    pub(crate) fn spawn_statement_logging_task(&self) {
+        let internal_cmd_tx = self.internal_cmd_tx.clone();
+        spawn(|| "statement_logging", async move {
+            // TODO[btv] make this configurable via LD?
+            // Although... Logging every 5 seconds seems like it
+            // should have acceptable cost for now, since we do a
+            // group commit for tables every 1s anyway.
+            let mut interval = tokio::time::interval(std::time::Duration::from_secs(5));
+            interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+            loop {
+                interval.tick().await;
+                let _ = internal_cmd_tx.send(Message::DrainStatementLog);
+            }
+        });
+    }
```

Comment:
This approach totally works, and to me looks good!

An alternative could be to push this logic into the write tasks for storage owned collections, and make them a bit more configurable. i.e. [write_task](https://github.com/ParkMyCar/materialize/blob/main/src/storage-client/src/controller/collection_mgmt.rs#L174-L178) already waits up to 1 second (`DEFAULT_TICK`) before writing to persist. You could could update write_task so a caller can provide tick and buffer intervals, and then the Coordinator holds onto appender channels. This is definitely an involved change so I'll leave it up to you!